### PR TITLE
Sdil 1485

### PR DESCRIPTION
**Pull request by adamdye-hmrc**

### CODE REVIEW

File: app/views/uniform/fragments/registration_not_required.scala.html [hunk 1]
```diff
@@ -0,0 +1,32 @@
...
+ *@
+
+@import sdil.config.AppConfig
+
+@()(implicit request: Request[_], messages: Messages, config: AppConfig)
+
+<p>@Messages("sdil.registration-not-required.p1")</p>
+
+<ul class="list-bullet">
+    <li>@Messages("sdil.registration-not-required.li1")</li>
```
Comment (hmrclt) [inline app/views/uniform/fragments/registration_not_required.scala.html]: Consider using `Messages.many`

File: app/views/uniform/fragments/variationsWHN.scala.html [hunk 2]
```diff
@@ -0,0 +1,23 @@
...
+ * Unless required by applicable law or agreed to in writing, software
+ * distributed under the License is distributed on an "AS IS" BASIS,
+ * WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+ * See the License for the specific language governing permissions and
+ * limitations under the License.
+ *@
+
+@import sdil.config.AppConfig
+ 
+@()(implicit messages: Messages, config: AppConfig)
```
Comment (hmrclt) [inline app/views/uniform/fragments/variationsWHN.scala.html]: is `config` used?

File: conf/messages [hunk 3]
```diff
@@ -484,6 +484,62 @@ delete=Remove
...
+heading.producer=How many litres of your own brands of liable drinks have been packaged globally in the past 12 months?
+producer.lead=This includes brands you own or have the rights to manufacture.
+producer.option.large=1 million litres or more
+producer.option.small=Less than 1 million litres
+producer.option.not=None
+
+producer.help.header.1=What does ’packaged’ mean?
+producer.help.1=A liable drink is packaged if it is bottled, canned or otherwise packaged so it’s ready to either drink or dilute.
+producer.help.header.2=What is a liable drink?
+producer.help.2=<h2>Liable drinks</h2>\
```
Comment (hmrclt) [inline conf/messages]: This really really needs to be in a view.

File: conf/messages [hunk 4]
```diff
@@ -564,6 +615,8 @@ variationDone.subheading=We’ve received the updates to your Soft Drinks Indust
 variationDone.what-happens.title=What happens next
 variationDone.what-happens-next=You don’t need to do anything else. We’ll update your details with the information \
 you provided. We’ll send you confirmation that the changes have been made.
+
+
```
Comment (hmrclt) [inline conf/messages]: why the break here?

File: app/sdil/config/RoutesWiring.scala [hunk 5]
```diff
@@ -24,7 +24,7 @@ import play.api.routing.Router
 import sdil.actions.{AuthorisedAction, FormAction, RegisteredAction}
 import sdil.connectors._
 import sdil.controllers.test.TestingController
-import sdil.controllers.{VariationsController => UniformVariationsController, _}
+import sdil.controllers.{VariationsController => UniformVariationsController, RegistrationController => UniformRegistrationController, _}
```
Comment (hmrclt) [inline app/sdil/config/RoutesWiring.scala]: line too long, consider either using two imports or cheating and calling them `UVariationsControllers` etc.

Comment (hmrclt) [reply app/sdil/config/RoutesWiring.scala]: As the fat forms are gone for variations the first aliasing might not be necessary anyway.

File: app/sdil/controllers/RegistrationController.scala [hunk 6]
```diff
@@ -0,0 +1,138 @@
...
+                          end("do-not-register",noReg)
+                        } else (()).pure[WebMonad]
+      regDate        <- ask(startDate
+                              .verifying("error.start-date.in-future", !_.isAfter(LocalDate.now))
+                              .verifying("error.start-date.before-tax-start", !_.isBefore(LocalDate.of(2018, 4, 6))), "start-date")
+      packSites       <- askPackSites(List.empty[Site], (packLarge.contains(true) && packageOwn.contains(true)) || !copacks.isEmpty)
+      isVoluntary     =  packLarge.contains(false) && useCopacker.contains(true) && (copacks, imports).isEmpty
+      warehouses      <- manyT("warehousesActivity", ask(warehouseSiteMapping,_)(warehouseSiteForm, implicitly)) emptyUnless !isVoluntary
+      contactDetails  <- ask(contactDetailsMapping, "contact-details")
+      activity        =  Activity(
```
[the remark this answers is not in the document]
Comment (adamdye-hmrc) [reply app/sdil/controllers/RegistrationController.scala]: Do you mean have a ask the questions used to compose the Activity in a separate function? Arguments for Activity's constructor are used elsewhere e.g. packLarge.

File: app/sdil/controllers/RegistrationController.scala [hunk 7]
```diff
@@ -0,0 +1,138 @@
...
+                           fd.rosmData.organisationName,
+                           orgType,
+                           UkAddress.fromAddress(fd.rosmData.address),
+                           activity,
+                           regDate,
+                           packSites,
+                           warehouses,
+                           contact
+                         )
+      declaration     =  uniform.fragments.declaration(
```
[the remark this answers is not in the document]
Comment (adamdye-hmrc) [reply app/sdil/controllers/RegistrationController.scala]: Returns & Variations both have their own CYA pages. Would be best to look at when refactoring.

File: app/sdil/controllers/SdilWMController.scala [hunk 8]
```diff
@@ -62,7 +63,8 @@ trait SdilWMController extends WebMonadController
   protected def askOneOf[A](
     id: String,
     possValues: List[A],
-    default: Option[A] = None
+    default: Option[A] = None,
+    sort: (((String, String),(String,String)) => Boolean) = _._2 < _._2
```
[the remark this answers is not in the document]
Comment (adamdye-hmrc) [reply app/sdil/controllers/SdilWMController.scala]: In radio list we sort the radio buttons by the labels, not the keys, _._1 is the key, _._2 is the label.
See uniform.fragments.radiolist...

File: app/sdil/controllers/RegistrationController.scala (hunk 6 above)
[the remark this answers is not in the document]
Comment (hmrclt) [reply app/sdil/controllers/RegistrationController.scala]: Still leaves a number of options for simplification. `packLarge` could be passed into the function, `Activity` could be amended to include the optional `packLarge`, the logic elsewhere might be able to work with the non-optional `packLarge`, or finally the function could return `WebMonad[(Activity, Option[Boolean)]`.

File: app/sdil/controllers/RegistrationController.scala (hunk 7 above)
[the remark this answers is not in the document]
Comment (hmrclt) [reply app/sdil/controllers/RegistrationController.scala]: Yes, `returnsCYA.scala.html` and `variationsCYA.scala.html`. I know what they do from the name. From `declaration.scala.html` I would assume it to be generic. It has been introduced in this branch so I would advise renaming it to something more appropriate as part of this work.

File: app/sdil/controllers/SdilWMController.scala (hunk 8 above)
[the remark this answers is not in the document]
Comment (hmrclt) [reply app/sdil/controllers/SdilWMController.scala]: So ... if you're not sorting by the key which is the `_._1` then my question still stands. Is there a situation where you might want to sort by the key?

**Pull request merged**